scipy/signal/lfilter.c.src

```diff
@@ -94,34 +95,40 @@ convert_shape_to_errmsg(npy_intp ndim, npy_intp *Xshape, npy_intp *Vishape,
         expect_size = j != theaxis ? Xshape[j] : val;
 
         if (j == ndim - 1) {
-            tmp = PyUString_FromFormat("%" NPY_INTP_FMT, expect_size);
-            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT, Vishape[j]);
+            tmp = PyUnicode_FromFormat("%" NPY_INTP_FMT, expect_size);
+            tmp1 = PyUnicode_FromFormat("%" NPY_INTP_FMT, Vishape[j]);
         } else {
-            tmp = PyUString_FromFormat("%" NPY_INTP_FMT ",", expect_size);
-            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT ",", Vishape[j]);
+            tmp = PyUnicode_FromFormat("%" NPY_INTP_FMT ",", expect_size);
+            tmp1 = PyUnicode_FromFormat("%" NPY_INTP_FMT ",", Vishape[j]);
         }
         if (!tmp) {
             Py_DECREF(msg);
             Py_DECREF(msg1);
             Py_XDECREF(tmp1);
             return 0;
         }
+        /* Should this be merged with above? */
         if (!tmp1) {
             Py_DECREF(msg);
             Py_DECREF(msg1);
             Py_DECREF(tmp);
             return 0;
         }
-        PyUString_ConcatAndDel(&msg, tmp);
-        PyUString_ConcatAndDel(&msg1, tmp1);
+        Py_SETREF(msg, PyUnicode_Concat(msg, tmp));
+        Py_SETREF(msg1, PyUnicode_Concat(msg1, tmp1));
+        Py_DECREF(tmp);
+        Py_DECREF(tmp1);
     }
-    tmp = PyUString_FromString(").");
+    tmp = PyUnicode_FromString(").");
     if (!tmp) {
+        /* Should not decref msg which gets returned? or should return 0? */
```

Comment:
It's not clear to me that tmp can be null here.

Reply:
Also in the other places with this form they all return 0
Is this block missing a return?

Reply:
Agreed, seems to be missing `return 0;`